dbms/src/DataStreams/WindowBlockInputStream.cpp

```diff
@@ -272,12 +274,120 @@ Int64 WindowTransformAction::getPartitionEndRow(size_t block_rows)
     return left;
 }
 
+RowNumber WindowTransformAction::stepForward(const RowNumber & current_row, Int64 n)
```

Comment:
```suggestion
RowNumber WindowTransformAction::stepForward(const RowNumber & current_row, UInt64 n)
```
Can `n` be negative?
If `UInt64` is used, the following `static_cast` can be removed

Reply:
> Can `n` be negative? If `UInt64` is used, the following `static_cast` can be removed

okk, I will change type to `UInt64`